postgres/datadog_checks/postgres/connections.py

```diff
@@ -51,12 +72,20 @@ def __init__(self, connect_fn):
                 )
         self.connect_fn = connect_fn
 
-    def get_connection(self, dbname, ttl_ms):
+    def get_connection(self, dbname: str, ttl_ms: int):
+        """
+        Grab a connection from the pool if the database is already connected.
+        If max_conns is specified, and the database isn't already connected,
+        make a new connection IFF the max_conn limit hasn't been reached.
+        If we can't fit the connection into the pool, return None.
+        """
         self.prune_connections()
         with self._mu:
-            conn = self._conns.pop(dbname, ConnectionWithTTL(None, None))
+            conn = self._conns.pop(dbname, ConnectionInfo(None, None, None, None, None))
             db = conn.connection
             if db is None or db.closed:
+                if self.max_conns is not None and len(self._conns) == self.max_conns:
+                    return None
```

Comment:
This puts the onus on the caller to implement a polling function and always check for a None value. From an ergonomic standpoint, I think it would be easier to build polling for an available connection directly into this function. What do you think about having `get_connection` block until a connection becomes available?

Reply:
I think positively about that!